compiler/ballerina-treegen/src/main/resources/syntax_tree_descriptor.json

```diff
@@ -4455,6 +4455,163 @@
+                    "type": "Node"
+                }
+            ]
+        },
+        {
+            "name": "ReCharacterClassNode",
+            "base": "ReAtomNode",
+            "kind": "RE_CHARACTER_CLASS",
+            "attributes": [
+                {
+                    "name": "openBracketAndNegation",
+                    "type": "Token"
+                },
+                {
+                    "name": "reCharSet",
+                    "type": "ReCharSetNode",
+                    "isOptional": true
+                },
+                {
+                    "name": "closeBracket",
+                    "type": "Token"
+                }
+            ]
+        },
+        {
+            "name": "ReCharSetNode",
+            "base": "Node",
+            "kind": "RE_CHAR_SET",
+            "attributes": [
+                {
+                    "name": "reCharSet",
+                    "type": "Node"
+                }
+            ]
+        },
+        {
+            "name": "ReCapturingGroupsNode",
+            "base": "ReAtomNode",
+            "kind": "RE_CAPTURING_GROUPS",
+            "attributes": [
+                {
+                    "name": "openParenthesis",
+                    "type": "Token"
+                },
+                {
+                    "name": "reFlagExpression",
+                    "type": "ReFlagExpressionNode",
+                    "isOptional": true
+                },
+                {
+                    "name": "reSequences",
+                    "type": "Node",
+                    "occurrences": "MULTIPLE"
+                },
+                {
+                    "name": "closeParenthesis",
+                    "type": "Token"
+                }
+            ]
+        },
+        {
+            "name": "ReFlagExpressionNode",
+            "base": "Node",
+            "kind": "RE_FLAG_EXPR",
+            "attributes": [
+                {
+                    "name": "questionMark",
+                    "type": "Token"
+                },
+                {
+                    "name": "reFlagsOnOff",
+                    "type": "ReFlagsOnOffNode"
+                },
+                {
+                    "name": "colon",
+                    "type": "Token"
+                }
+            ]
+        },
+        {
+            "name": "ReFlagsOnOffNode",
+            "base": "Node",
+            "kind": "RE_FLAGS_ON_OFF",
+            "attributes": [
+                {
+                    "name": "reFlagsOnOff",
+                    "type": "Node"
+                }
+            ]
+        },
+        {
+            "name": "ReAssertionNode",
+            "base": "ReTermNode",
+            "kind": "RE_ASSERTION",
+            "attributes": [
+                {
+                    "name": "reAssertion",
+                    "type": "Node"
+                }
+            ]
+        },
+        {
+            "name": "ReQuantifierNode",
```

Comment:
`ReQuantifier := ReBaseQuantifier ["?"]` Don't we need to have optional `?` separately for this node?